samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/SubmitToGradingActionListener.java

```diff
@@ -445,6 +452,12 @@ private AssessmentGradingData persistAssessmentGrading(ActionEvent ae,
 				log.debug("Submitforgrading: newItemGradingSet.size = {}", adds.size());
 
 				HashSet<ItemGradingData> updateItemGradingSet = getUpdateItemGradingSet(itemGradingSet, adds, calcQuestionMap,imagQuestionMap, emiMap, adata);
+				if (!studentRichText) {
+					updateItemGradingSet.stream()
+					.filter(igd -> essayShortAnwerQuestionMap.containsKey(igd.getPublishedItemId()))
+					.collect(Collectors.toMap(Function.identity(), igd -> ComponentManager.get(FormattedText.class).escapeHtml(igd.getAnswerText(), false)))
```

Comment:
This is not an optimal use of the ComponentManager in a lambda. This should be initialized elsewhere i.e. in the class constructor.

Reply:
Done